rfcs/0000-address-format/0000-address-format.md

```diff
@@ -0,0 +1,105 @@
+---
+Number: 21
+Category: Standards Track
+Status: Proposal
+Author: Cipher Wang
+Organization: Nervos Foundation
+Created: 2019-01-20
+---
+
+# Address Format
+
+## Abstract
+
+CKB Address Format is an application level lock script display recommendation. The lock script consists of two key parameters, including *code hash* and *arguments*. In the consideration of user experience, it is necessary to wrap the raw data structure into a verifiable and extensible format.
+
+## Solution
+
+CKB Address Format follows [Bitcoin base32 address format (BIP-173)][bip173] rules, which wraps code_hash ahd args in **Bech32** encoding and a [BCH checksum][bch].
+
+A Bech32 string is at most 90 characters long and consists of the **human-readable part**, the **separator**, and the **data part**. The last 6 characters of data part is checksum. The data part is base32 encoded. Here is the readable translation of base32 encoding table.
+
+|0|1|2|3|4|5|6|7|
+|-|-|-|-|-|-|-|-|
+|**+0**|q|p|z|r|y|9|x|8|
+|**+8**|g|f|2|t|v|d|w|0|
+|**+16**|s|3|j|n|5|4|k|h|
+|**+24**|c|e|6|m|u|a|7|l|
+
+
+The human-readable part is "**ckb**" for CKB mainnet, and "**ckt**" for the testnet. The separator is always "1".
+
+![](images/ckb-address.png)
+
+## Payload
+
+The first step to encode lock script into address is to encode it to payload. We use type field in payload to identify different encoding methods according to different user scenario needs, and parameter fields to represent lock script data.
+
+```
+payload = type | parameter1 | parameter2 | ...
+```
+
+|   type     |    parameter1    | parameter2  | => lock script |
+|------------|------------------|-------------|-------------|
+|    0x01    | 1 byte index |  PK/PKHash  | {code_hash: libs[p1], args:[p2]} |
+|    other   | reserved  |  --   | -- |
+
+Type 1 is a compact address format which identifies common used code hash by 1 byte code hash index instead of 32 bytes code hash. Other type number address formats are reserved.
```

Comment:
'reserved' is not accurate. Other types are not defined yet and can be added in future.